src/string_immutable.zig

```diff
@@ -3457,15 +3482,15 @@ pub fn firstNonASCII16CheckMin(comptime Slice: type, slice: Slice, comptime chec
     }
 
     if (comptime check_min) {
-        for (remaining) |char| {
+        for (remaining) |char, i| {
             if (char > 127 or char < 0x20) {
-                return @truncate(u32, (@ptrToInt(std.mem.sliceAsBytes(remaining).ptr) - @ptrToInt(std.mem.sliceAsBytes(slice).ptr)) / 2);
```

Comment:
Was there a bug here or was the code confusing? Adding `i` to this code makes the codegen worse

Reply:
There's a bug, this won't return the correct index of the non-ascii character. How bad is codegen with `i`? I could add a counter variable instead